### PR TITLE
5.9 Windows macros support changes

### PR DESCRIPTION
**Pull request by compnerd**
This backports the changes to support macros on Windows.  The bulk of the changes are to the tests, the C++ and runtime pieces fix mismatches between `operator new`/`malloc` and `free` and implement Windows process execution support.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @bnbarham @rintaro @DougGregor @shahmishal

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7399

@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7399

@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7399

@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7399

@swift-ci please test Linux platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7399

@swift-ci please test macOS platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7399

@swift-ci please test macOS platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7399

@swift-ci please test macOS platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7404
https://github.com/apple/swift-syntax/pull/2161
https://github.com/apple/swift-driver/pull/1435

@swift-ci please test

**Review by DougGregor — APPROVED** [review]
This looks great. Lots of changes, but we've been vetting these on main already, and they are mainly focused in portability (so low risk) or are limited to relatively new code for the Windows build. Thank you!

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7404
https://github.com/apple/swift-syntax/pull/2161
https://github.com/apple/swift-driver/pull/1435

@swift-ci please test

File: tools/swift-plugin-server/Sources/CSwiftPluginServer/include/PluginServer.h [hunk 1]
```diff
@@ -13,6 +13,9 @@
 #ifndef SWIFT_PLUGINSERVER_PLUGINSERVER_H
 #define SWIFT_PLUGINSERVER_PLUGINSERVER_H
 
+#include <stddef.h>
+#include <stdint.h>
+
```
Comment (bnbarham) [inline tools/swift-plugin-server/Sources/CSwiftPluginServer/include/PluginServer.h:18]: We need https://github.com/apple/swift/commit/4cb8133e4c0025473c09574a18f1a3f617fa0d6c as well I believe.

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/llvm-project/pull/7404
https://github.com/apple/swift-syntax/pull/2161
https://github.com/apple/swift-driver/pull/1435

@swift-ci please test

**Pull request merged**